### PR TITLE
Fix OADP-5184: Add doc for automount_service_account_token exception

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**
## Why the changes were made

Documentation in order to comply with the exception process: https://issues.redhat.com/browse/OADP-5184 

## How to test the changes made

Review the document added in the PR

### CODE REVIEW

File: docs/automount_service_account_token_exception.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+- Without this token, the operator cannot perform any of its intended functions, such as managing resources or reconciling the desired state.
+
+### **2. Alignment with Kubernetes Operator Design**
+- The OADP Operator relies on the standard Kubernetes operator design, which assumes that service account tokens are automatically mounted into pods.
+- Rewriting the operator to use alternative authentication methods (e.g., manually injected secrets) would introduce significant complexity and deviate from best practices for operator development.
+
+### **3. Maintaining Backup and Restore Functionality**
+- The OADP Operator facilitates critical backup and restore operations for workloads in the cluster.
+- Disabling `automountServiceAccountToken` would result in failed operations.
+
```
Comment (kaovilai) [inline docs/automount_service_account_token_exception.md:49]:
```suggestion
- Short Term Service Account tokens are rotated externally via cloud-credentials-operator requires us to use the automount token to authenticate against cloud providers.
```

**Review by kaovilai — CHANGES_REQUESTED** [review]

File: docs/automount_service_account_token_exception.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# **Knowledge Base Article: Why the OADP Operator Pod Requires `automountServiceAccountToken` to Be Enabled**
+
+---
+
+## **Rationale for Allowing the Default `automountServiceAccountToken` Behavior in OADP Operator Pods**
+
+The Kubernetes Best Practices team recommends setting `automountServiceAccountToken: false` for pods to reduce security risks by minimizing exposure of service account tokens. However, the **OADP Operator Pod** relies on the Kubernetes API server for its critical functions, and disabling the token mount would render it inoperable. While the OADP Operator Pod does not explicitly set `automountServiceAccountToken: true`, it depends on the **default Kubernetes behavior**, which mounts the service account token into the pod.
```
Comment (mateusoliveira43) [inline docs/automount_service_account_token_exception.md:7]: It is not clear for me which Pod is `OADP Operator Pod`?

Comment (mateusoliveira43) [inline docs/automount_service_account_token_exception.md:7]: Is not better to explicitly set it true to avoid this changing and breaking in the future?

**Review by weshayutin — COMMENTED** [review]
Thank you @shubham-pampattiwar for working through these.  Couple nits and we'll approve

File: docs/automount_service_account_token_exception.md (hunk 2 above)
Comment (weshayutin) [reply docs/automount_service_account_token_exception.md:7]: We are required to document why it's not false, nothing more https://issues.redhat.com//browse/OADP-5184  IMHO less is better.

Comment (weshayutin) [reply docs/automount_service_account_token_exception.md:7]: let's use the actual pod name as well.

Comment (shubham-pampattiwar) [reply docs/automount_service_account_token_exception.md:7]: yeah I would not make any changes/try to fix whats not broken.

File: docs/automount_service_account_token_exception.md (hunk 1 above)
Comment (shubham-pampattiwar) [reply docs/automount_service_account_token_exception.md:50]: Done, updated the PR

File: docs/automount_service_account_token_exception.md (hunk 2 above)
Comment (shubham-pampattiwar) [reply docs/automount_service_account_token_exception.md]: Done, updated the PR

**Review by weshayutin — APPROVED** [review]
/LGTM

**Review by kaovilai — APPROVED** [review]

**Pull request merged**